test/kubernetes/e2e/features/rate_limit/global/suite.go

```diff
@@ -0,0 +1,268 @@
+package global_rate_limit
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"github.com/stretchr/testify/suite"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/kgateway-dev/kgateway/v2/pkg/utils/kubeutils"
+	"github.com/kgateway-dev/kgateway/v2/pkg/utils/requestutils/curl"
+	testmatchers "github.com/kgateway-dev/kgateway/v2/test/gomega/matchers"
+	"github.com/kgateway-dev/kgateway/v2/test/kubernetes/e2e"
+	testdefaults "github.com/kgateway-dev/kgateway/v2/test/kubernetes/e2e/defaults"
+)
+
+var _ e2e.NewSuiteFunc = NewTestingSuite
+
+// testingSuite is a suite of global rate limiting tests
+type testingSuite struct {
+	suite.Suite
+
+	ctx context.Context
+
+	// testInstallation contains all the metadata/utilities necessary to execute a series of tests
+	// against an installation of kgateway
+	testInstallation *e2e.TestInstallation
+
+	// manifests shared by all tests
+	commonManifests []string
+	// resources from manifests shared by all tests
+	commonResources []client.Object
+	agentGateway    bool
```

Comment:
[nit] Do we want to follow `agentgateway` or `agentGateway everywhere? 
```suggestion
	agentgateway    bool
```